File: app/src/main/java/com/algorithmers/android/algoapp/AlgoMainActivity.java
package com.algorithmers.android.algoapp;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class AlgoMainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);


        this.setContentView(R.layout.activity_algo_main);


        SharedPreferences sharedPreferences = this.getSharedPreferences("USER_AUTH", MODE_PRIVATE);


        String email = sharedPreferences.getString("Email", "Empty_Email");

        String password = sharedPreferences.getString("Password", "Empty_Password");


        if (!email.equals("Empty_Email") || !password.equals("Empty_Password")) {


            this.startActivity(new Intent(this, AlgoSnippetListActivity.class));

            this.finishAffinity();

        }


    }


    public void navigator(View v) {

        if (v.getId() == R.id.btnActMainSignIn) {


            this.startActivity(new Intent(this, AlgoSignInActivity.class));

        } else {


        }

    }
}

File: app/src/main/java/com/algorithmers/android/algoapp/AlgoMainActivity.kt
package com.algorithmers.android.algoapp

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View

class AlgoMainActivity : Activity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.setContentView(R.layout.activity_algo_main)
    }

    fun nav(view: View?) {
        val viewId = view!!.id
        if (viewId == R.id.btnActMainSignIn) {
            this.startActivity(Intent(this, AlgoMainServicesActivity::class.java))
            this.finishAffinity()
        } else if (viewId == R.id.btnActMainForgetPassword) {

        } else if (viewId == R.id.btnActMainSignUp) {
            this.startActivity(Intent(this, AlgoSignUpActivity::class.java))
        }
    }
}
